website/docs/configuration/modules.html.md

```diff
@@ -354,73 +447,41 @@ version constraints for a particular child module without creating a local
 module configuration, use the [`required_providers`](/docs/configuration/terraform.html#specifying-required-provider-versions)
```

Comment:
This is outside of the area you've changed, but I find the phrasing in this paragraph a bit off. We're using the phrase  "without creating a local module configuration" where (I believe) we really mean "without creating a [local] provider configuration within the module". 
I don't feel super strongly about this, so feel free to ignore this bit of feedback, but I noticed it when reading through the rendered version of this PR.

Reply:
Oh yes, I agree... the terminology here doesn't really match with the terminology elsewhere on the page and so it's not immediately clear what it's talking about.

Since the `version` argument for `provider` blocks is already documented as legacy in the Providers page (because `required_providers` has replaced it since 0.12), I took this opportunity to revise it to simply state not to use it, rather than to discuss specific implications of doing so.